netbox/netbox.py

```diff
@@ -129,28 +130,23 @@ def _get_value_by_path(self, source_dict, key_path,
                 sys.exit("The key %s is not found. Please remember, Python is case sensitive." % key_name)
         return key_value
 
-    def _config(self, key_path, default=""):
+    def _config(self, key_path, default="", ignore_key_error=False):
         """Get value from config var.
 
         Args:
             key_path: A list has the path of the key.
             default: Default value if the key is not found.
 
         Returns:
-            The value of the key from config file or the default value.
+            The value of the key from config file or the default value
         """
         config = self.script_config.setdefault("netbox", {})
-        value = self._get_value_by_path(config, key_path, ignore_key_error=True, default=default)
-
-        if value:
-            key_value = value
-        else:
-            sys.exit("The key '%s' is not found in config file." % ".".join(key_path))
+        key_value = self._get_value_by_path(config, key_path, ignore_key_error=ignore_key_error, default=default)
```

Comment:
I kept that part like that because I like to have a different message than the default one in `_get_value_by_path` (I found it a bit misleading, that's why I use a different error message here).
So I'd suggest to edit `_get_value_by_path` to have the ability to change the error message.

Reply:
This is the part that would result in sys.exit() if a default value of "" was assigned to api_token when missing from netbox.yml.

I'm reviewing the logic to see if there's a simpler/better way to have an optional key for api_token in netbox.yml.